java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/MacAddress.java

```diff
@@ -202,6 +202,27 @@ public void putTo(PrimitiveSink sink) {
         sink.putShort((short) (this.rawValue & 0xFFFF));
     }
 
-
-
+    /** Parse an IPv4 Multicast address and return the macAddress
+     * corresponding to the multicast IPv4 address.
+     * 
+     * @param ipv4 - ipv4 multicast address
+     * @return the MacAddress corresponding to the multicast address
+     * @throws IllegalArgumentException if ipv4 is not a valid multicast address
+     */
+    @Nonnull
+    public static MacAddress forIPv4MulticastAddress(IPv4Address ipv4)
```

Comment:
Maybe add some code that explains how you're transplanting the lower 23 bits to the mac address.